src/core/operations_problem_results.jl

```diff
@@ -1,48 +1,107 @@
     base_power::Float64
     variable_values::Dict{Symbol, DataFrames.DataFrame}
-    total_cost::Dict
-    optimizer_log::Dict
-    time_stamp::DataFrames.DataFrame
     dual_values::Dict{Symbol, DataFrames.DataFrame}
     parameter_values::Dict{Symbol, DataFrames.DataFrame}
+    optimizer_stats::OptimizerStats
+    output_dir::String
 end
 
 get_existing_variables(res::OperationsProblemResults) = keys(get_variables(res))
 get_existing_parameters(res::OperationsProblemResults) = keys(IS.get_parameters(res))
 get_existing_duals(res::OperationsProblemResults) = keys(get_duals(res))
 get_model_base_power(res::OperationsProblemResults) = res.base_power
+get_objective_value(res::OperationsProblemResults) = res.optimizer_stats.objective_value
 IS.get_variables(res::OperationsProblemResults) = res.variable_values
 IS.get_total_cost(res::OperationsProblemResults) = res.total_cost
-IS.get_optimizer_log(res::OperationsProblemResults) = res.optimizer_log
+IS.get_optimizer_stats(res::OperationsProblemResults) = res.optimizer_stats
 IS.get_timestamp(res::OperationsProblemResults) = res.time_stamp
 get_duals(res::OperationsProblemResults) = res.dual_values
 IS.get_parameters(res::OperationsProblemResults) = res.parameter_values
 
-function OperationsProblemResults()
-    #vars_result = read_variables(op_problem)
-    #param_values = read_parameters(op_problem)
-    #optimizer_log = get_optimizer_log(op_problem)
-    #time_stamp = get_timestamps(op_problem)
-    #time_stamp = shorten_time_stamp(time_stamp)
-    #base_power = PSY.get_base_power(op_problem.sys)
-    #dual_result = read_duals(op_problem)
-    #obj_value = Dict(
-    #    :OBJECTIVE_FUNCTION => JuMP.objective_value(op_problem.psi_container.JuMPmodel),
-    #)
-    #base_power = get_model_base_power(op_problem)
-    #merge!(optimizer_log, timed_log)
-
-    #results = OperationsProblemResults(
-    #    base_power,
-    #    vars_result,
-    #    obj_value,
-    #    optimizer_log,
-    #    time_stamp,
-    #    dual_result,
-    #    param_values,
-    #)
+function OperationsProblemResults(problem::OperationsProblem)
+    status = get_run_status(problem)
+    status != RunStatus.SUCCESSFUL && error("problem was not solved successfully: $status")
+
+    container = get_optimization_container(problem)
+    variables = read_variables(container)
+    duals = read_duals(container)
+    parameters = read_parameters(container)
+    timestamps = get_timestamps(problem)
+    optimizer_stats = OptimizerStats(problem)
+
+    for df in Iterators.flatten(((values(variables), values(duals), values(parameters))))
+        DataFrames.insertcols!(df, 1, :DateTime => timestamps)
+    end
+
+    return OperationsProblemResults(
+        get_problem_base_power(problem),
+        variables,
+        duals,
+        parameters,
+        optimizer_stats,
+        mkpath(joinpath(get_output_dir(problem), "results")),
+    )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", results::OperationsProblemResults)
```

Comment:
should we move this to the print.jl file?

Reply:
I'm not sure where our convention started. Is it a Julia thing? I personally feel that it's natural to put the show method near the struct.

Reply:
Most Julia packages we use, use this pattern so we adopted it.